provisioner/node_pools.go

```diff
@@ -237,14 +237,47 @@ func (p *KarpenterNodePoolProvisioner) Reconcile(ctx context.Context, _ updatest
 		return err
 	}
 	for _, pr := range existingProvisioners.Items {
-		if !inNodePoolList(&api.NodePool{Name: pr.GetName()}, karpenterPools) {
+		nodePool := api.NodePool{Name: pr.GetName(), Profile: karpenterNodePoolProfile}
+		if !inNodePoolList(&nodePool, karpenterPools) {
 			err := p.k8sClients.Delete(ctx, crdResolver.NodePoolCRDName, "", pr.GetName(), metav1.DeleteOptions{})
 			if err != nil {
 				return err
 			}
 		}
 	}
 
+	nodes, err := p.k8sClients.List(ctx, "nodes", "", metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("karpenter.sh/nodepool"),
+	})
+	if err != nil {
+		return err
+	}
+
+	// also label nodes for decommissioning so CLC can speed up the decommissioning process,
+	// enforcing our SLAs since karpenter will wait indefinitely for the pod disruption budgets
+	// https://cloud.docs.zalando.net/reference/cluster-update-policy/#slas
```

Comment:
nit: This depends on the updatestrategy used which for us is always CLC but could be the CLM one as well, so maybe we don't explicit mention CLC here?

Also we should not link to internal docs in a comment here.

Reply:
updated